src/lib.rs

```diff
@@ -391,38 +401,36 @@ impl Trajectory for TRRTrajectory {
             // Similar for lambda.
             let code = xdrfile_trr::read_trr(
                 self.handle.xdrfile,
-                num_atoms as i32,
+                to_i32(num_atoms, ErrorTask::Read)?,
                 &mut step,
                 &mut frame.time,
                 &mut lambda,
                 &mut frame.box_vector,
                 frame.coords.as_mut_ptr(),
                 std::ptr::null_mut(),
                 std::ptr::null_mut(),
-            ) as u32;
-
-            frame.step = step as usize;
+            );
             if let Some(err) = check_code(code, ErrorTask::Read) {
-                Err(err)
-            } else {
-                Ok(())
+                return Err(err);
             }
+            frame.step = usize::try_from(step).map_err(|_| Error::StepSizeOutOfRange(step))?;
+            Ok(())
```

Comment:
I think it would be the appropriate to handle this similar an error when reading num atoms instead of a separate error type.

I.e:
```rust
expect("Encountered illegal value for frame when writing trajectory: Failed to cast {} to usize.", step). 
```

Reply:
I guess I did it the other way around! But I've clarified the error message